src/fwriteR.c

```diff
@@ -5,18 +5,23 @@
 #define DATETIMEAS_EPOCH     2
 #define DATETIMEAS_WRITECSV  3
 
+#define NEED2NATIVE(s) !(IS_ASCII(s) || (s)==NA_STRING)
+#define ENCODED_CHAR(s) (utf8 && NEED2UTF8(s) ? translateCharUTF8(s) : \
+  (native && NEED2NATIVE(s)) ? translateChar(s) : CHAR(s))
```

Comment:
I don't get parenthesis here... does it work as expected and I am just unable to read it properly? :/
why not like this?
```
(utf8 && NEED2UTF8(s)) ? translateCharUTF8(s) : ((native && NEED2NATIVE(s)) ? translateChar(s) : CHAR(s))
```

Reply:
I just think it's too long to fit in a single line... nothing special. I'm ok to put it to one line.

Reply:
it is not just single line but my option has different parentheses, two lines are no problem

Reply:
ah ok, I think I know how that works. It just doesn't have "else" condition into new parentheses. I would personally add those parentheses so it is easier to see branches in this ternary operator. And multiline is fine.